EssentialsDiscord/src/main/java/net/essentialsx/discord/listeners/DiscordListener.java

```diff
@@ -105,19 +106,21 @@ public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
             }
         }
 
+        Iterable<com.earth2me.essentials.User> viewers = plugin.getPlugin().getEss().getOnlineUsers();
         // Do not create the event specific objects if there are no listeners
         if (DiscordRelayEvent.getHandlerList().getRegisteredListeners().length != 0) {
             final DiscordRelayEvent relayEvent = new DiscordRelayEvent(
                     new InteractionMemberImpl(member), new InteractionChannelImpl(event.getChannel()),
-                    Collections.unmodifiableList(keys), event.getMessage().getContentRaw(), formattedMessage);
+                    Collections.unmodifiableList(keys), event.getMessage().getContentRaw(), formattedMessage, Lists.newArrayList(viewers));
```

Comment:
This set of viewers should really be a set of viewers who have the receive permissions for the channel we're sending to. Move the permission check from 125-127 up to here to get the list of users